src/tasks/JsonToItemsTaskFactory/JsonToItemsTaskFactory.cs

```diff
@@ -0,0 +1,379 @@
+        }
+
+        public class JsonToItemsTask : IGeneratedTask
+        {
+            private IBuildEngine? _buildEngine;
+            public IBuildEngine BuildEngine { get => _buildEngine!; set { _buildEngine = value; SetBuildEngine(value);} }
+            public ITaskHost? HostObject { get; set; }
+
+            private TaskLoggingHelper? _log;
+            private TaskLoggingHelper Log { get => _log!; set { _log = value; } }
+
+            private void SetBuildEngine (IBuildEngine buildEngine)
+            {
+                Log = new TaskLoggingHelper(buildEngine, TaskName);
+            }
+
+            public static JsonSerializerOptions JsonOptions => new()
+                        {
+                            PropertyNameCaseInsensitive =  true,
+                            AllowTrailingCommas = true,
+                        };
+            private string? jsonFilePath;
+
+            private readonly bool logDebugTask; // print stuff to the log for debugging the task
+
+            private JsonModelRoot? jsonModel;
+            public string TaskName {get;}
+            public JsonToItemsTask(string taskName, bool logDebugTask = false)
+            {
+                TaskName = taskName;
+                this.logDebugTask = logDebugTask;
+            }
+
+            public bool Execute()
+            {
+                if (jsonFilePath == null)
+                {
+                    Log.LogError("no JsonFilePath specified");
+                    return false;
+                }
+                if (!TryGetJson (jsonFilePath, out var json))
+                    return false;
+
+                if (logDebugTask)
+                {
+                    LogParsedJson (json);
+                }
+                jsonModel = json;
+                return true;
+            }
+
+            public bool TryGetJson(string jsonFilePath, [NotNullWhen(true)] out JsonModelRoot? json)
+            {
+                FileStream? file = null;
+                try
+                {
+                    try
+                    {
+                        file = File.OpenRead(jsonFilePath);
+                    }
+                    catch (FileNotFoundException fnfe)
+                    {
+                        Log.LogErrorFromException(fnfe);
+                        json = null;
+                        return false;
+                    }
+                    json = JsonSerializer.DeserializeAsync<JsonModelRoot>(file, JsonOptions).AsTask().Result;
+                    if (json == null)
+                    {
+                        Log.LogError ("Failed to deserialize Json file");
+                        return false;
+                    }
+                    return true;
+                }
+                finally
+                {
+                    if (file != null)
+                        file.Dispose();
+                }
+            }
+
+            internal void LogParsedJson (JsonModelRoot json)
```

Comment:
Maybe the log messages from this should use `MessageImportance.Low`, instead of the default `MessageImportance.Normal`? and same for other debug messages